proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -617,13 +637,15 @@ ProToken plusMinusStart(int inputType) throws IOException {
   }
 
   ProToken periodStart() throws IOException {
+    LOGGER.trace("Entering periodStart()");
+
     if (!Character.isDigit(currChar)) {
       if (prepro.isNameDot())
-        return makeToken(ProParserTokenTypes.NAMEDOT);
+        return makeToken(PreprocessorParser.NAMEDOT);
       else
-        return makeToken(ProParserTokenTypes.PERIOD);
+        return makeToken(PreprocessorParser.PERIOD);
     }
-    int ttype = ProParserTokenTypes.NUMBER;
+    int ttype = PreprocessorParser.NUMBER;
     for_loop : for (;;) {
       switch (currChar) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of switch cases from 46 to at most 30. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1479)